### PR TITLE
EFA-397 Localize Suspicious folder displayed name

### PR DESCRIPTION
**Pull request by ig-garcia**

### CODE REVIEW

File: k9mail/src/main/res/values/pep_strings.xml [hunk 1]
```diff
@@ -459,5 +459,6 @@
     <string name="mdm_account_removed_dialog_description">Your IT department removed some accounts of this device. App needs to be restarted.</string>
     <string name="mdm_wrong_account_settings_dialog_title">Wrong account settings received</string>
     <string name="mdm_wrong_account_settings_dialog_description">It seems like your IT department sent settings to install new accounts but something is wrong. Please contact your IT admin.</string>
+    <string name="special_mailbox_name_suspicious">Suspicious</string>
```
Comment (Stereodark) [inline k9mail/src/main/res/values/pep_strings.xml:462]: so in the build for a language we will replace this xml with one that has the translations. Makes sense

**Review by Stereodark — APPROVED** [review]
looks good, maybe we should add to the commit message "migration FolderInfoHolder from Java to KT" or something similar as when I read the ticket I thought it was just the localization change, but it's also the Kotlin migration.

File: k9mail/src/main/java/com/fsck/k9/activity/FolderInfoHolder.kt [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        return if (ret != 0) {
+            ret
+        } else {
+            s1.compareTo(s2)
+        }
+    }
+
+    private fun truncateStatus(message: String?): String? {
+        var mess = message
+        if (mess != null && mess.length > 27) {
```
Comment (pzplanck) [inline k9mail/src/main/java/com/fsck/k9/activity/FolderInfoHolder.kt:63]: magic number should be dedicated constant. complexity case

File: k9mail/src/main/java/com/fsck/k9/activity/FolderInfoHolder.kt [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            context: Context,
+            account: Account,
+            name: String?,
+            fqn: String?
+        ): String? {
+            // FIXME: We really shouldn't do a case-insensitive comparison here
+            return when {
+                fqn == account.spamFolderName ->
+                    String.format(
+                        context.getString(R.string.special_mailbox_name_spam_fmt), name
```
Comment (pzplanck) [inline k9mail/src/main/java/com/fsck/k9/activity/FolderInfoHolder.kt:133]: that is good on the paper, but in reality of the actual project simply use K9.app.getString() for accessing resources. 
it is available while app is alive which can get rid of the NULL checks in constructor for the context which might disappear due to the activity/fragment lifecycle.

**Review by pzplanck — APPROVED** [review]

**Pull request merged**